doozer/doozerlib/backend/konflux_client.py

```diff
@@ -524,14 +492,18 @@ def build_pipeline_url(pipelinerun):
                 f"applications/{application}/"
                 f"pipelineruns/{pipelinerun_name}")
 
-    async def wait_for_pipelinerun(self, pipelinerun_name: str, namespace: Optional[str] = None) -> (resource.ResourceInstance, List[Dict]):
+    async def wait_for_pipelinerun(self, pipelinerun_name: str, namespace: Optional[str] = None, timeout_timedelta: Optional[datetime.timedelta] = None) -> (resource.ResourceInstance, List[Dict]):
```

Comment:
```suggestion
    async def wait_for_pipelinerun(self, pipelinerun_name: str, namespace: Optional[str] = None, timeout_timedelta: Optional[datetime.timedelta] = datetime.timedelta(hours=1)) -> (resource.ResourceInstance, List[Dict]):
```

Reply:
Well I assumed that programmers by default will look at the function definitions to understand what are the accepted values first, before they pass in the values. Since if we want to prevent errors like this, we should be adopting even more rigorous checks right